### PR TITLE
added 2.2

### LINKED ISSUE #20: Translate section 2.2
[Limits and Colimits](https://bartoszmilewski.com/2015/04/15/limits-and-colimits/)

### PR DESCRIPTION
**Pull request by kailuowang**
fixes #20 
Seeking feedback at a couple of places. See below

### CODE REVIEW

File: src/main/tut/2.2-limits-and-colimits.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+```
+```scala
+val βA = αA compose f 
+```
+..................
+```Haskell
+c -> Nat(Δc, D)
+```
+```scala
+C => (Δc ~> D)
```
Comment (kailuowang) [inline src/main/tut/2.2-limits-and-colimits.md]: translation of `Nat` (natural transformation) needs to be consistent with previous sections.

File: src/main/tut/2.2-limits-and-colimits.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+```
+
+..................
+```Haskell
+p'() = (0, 0)
+```
+```scala
+def p'[U](u: U) = u match {
+  case _: Unit => (0, 0) 
+}
```
Comment (kailuowang) [inline src/main/tut/2.2-limits-and-colimits.md]: Not too sure about this part. Is there a better way?

Comment (SystemFw) [reply src/main/tut/2.2-limits-and-colimits.md]: Right, the fact that  `Unit`, the empty parameter list and the singleton Tuple are 3 different things in scala, is very annoying. The translation above is incorrect however I think: partial, and non parametric.

How about:
```scala
def p: Unit => (Int, Int) = _ => (0,0)
```

**Comment by kailuowang** [comment]
addressed the feedback. Also I found a unicode prime symbol that works with scalac

**Pull request merged**